lib/MySQL_Protocol.cpp

```diff
@@ -2090,15 +2092,15 @@ bool MySQL_Protocol::process_pkt_handshake_response(unsigned char *pkt, unsigned
 #endif // debug
 						(*myds)->sess->default_hostgroup=default_hostgroup;
 						(*myds)->sess->default_schema=default_schema; // just the pointer is passed
-						(*myds)->sess->user_attributes = attributes; // just the pointer is passed , but for now not available in LDAP
+						(*myds)->sess->user_attributes = attributes; // just the pointer is passed, LDAP returns empty string
 #ifdef DEBUG
 						debug_spiffe_id(user,attributes, __LINE__, __func__);
 #endif
 						(*myds)->sess->schema_locked=schema_locked;
 						(*myds)->sess->transaction_persistent=transaction_persistent;
 						(*myds)->sess->session_fast_forward=fast_forward;
 						(*myds)->sess->user_max_connections=max_connections;
-						if (strncmp(password,(char *)pass,strlen(password))==0) {
+						if (strcmp(password, (char *) pass) == 0) {
```

Comment:
https://stackoverflow.com/questions/24353504/whats-wrong-with-strcmp

Reply:
Hi, can you please formulate?
Thanks